codersdk/workspaces.go

```diff
@@ -33,6 +33,26 @@ type Workspace struct {
 	LastUsedAt        time.Time      `json:"last_used_at"`
 }
 
+// OrphanTerraformState removes all the resources from the provided Terraform
+// state. When the new state is used, Terraform will operate as if none
+// of the resources in the original state exist.
+func OrphanTerraformState(state []byte) ([]byte, error) {
```

Comment:
Why not just use `{}`?

Reply:
Terraform has some requirements. E.g

<img width="1533" alt="Screen Shot 2022-09-02 at 5 01 38 PM" src="https://user-images.githubusercontent.com/7416144/188240707-32bd9149-3745-4158-83a1-5ccdcaa56cc7.png">

Instead of hard-coding these, I figured it was more future proof to just empty the resources array.

Reply:
Could you use a zero-byte value then? I don't like having to support potentially multiple versions of the Terraform statefile. We don't do this even when parsing right now, because we call the Terraform CLI.

Reply:
No. Terraform rejects state files that don't have a "version" attributes.